snippets/base/templates/base/includes/snippet_js.html

```diff
@@ -72,20 +73,26 @@
             show_snippet_id = metaParent.dataset.snippetId || 'unknown';
             show_snippet_campaign = metaParent.dataset.campaign || '';
         }
-        document.getElementById('snippet_set').setAttribute('data-snippet-id', show_snippet_id);
-        document.getElementById('snippet_set').setAttribute('data-campaign', show_snippet_campaign);
+        snippet_set.setAttribute('data-snippet-id', show_snippet_id);
+        snippet_set.setAttribute('data-campaign', show_snippet_campaign);
         sendImpression();
 
         // Trigger show_snippet event on snippet node.
         var evt = document.createEvent('Event');
         evt.initEvent('show_snippet', true, true);
-        show_snippet.dispatchEvent(evt);
 
-        // Add sample rate and snippet ID to currently displayed links.
-        var parameters = ('sample_rate=' + SNIPPET_METRICS_SAMPLE_RATE + '&snippet_name=' +
-                          show_snippet_id);
-        modifyLinks(show_snippet.getElementsByTagName('a'), parameters);
-        addSnippetBlockLinks(show_snippet.getElementsByClassName('block-snippet-button'));
+        // By the time show_snippet event reaches snippet_set the snippet
+        // will have finished initializing and altering the DOM. It's the
+        // time to modifyLinks() and addSnippetBlockLinks().
+        snippet_set.addEventListener('show_snippet', function(event) {
+            var topSection = document.getElementById('topSection');
+            // Add sample rate and snippet ID to currently displayed links.
+            var parameters = ('sample_rate=' + SNIPPET_METRICS_SAMPLE_RATE + '&snippet_name=' +
+                              show_snippet_id);
+            modifyLinks(topSection.getElementsByTagName('a'), parameters);
```

Comment:
No, not Fx only, all modern browsers support it and yes, unless you are doing `getElementById` it is faster.